model/lifecycle.go

```diff
@@ -1314,6 +1300,71 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 	return t, nil
 }
 
+func getDistrosFromRunOn(id string, buildVarTask BuildVariantTaskUnit, buildVariant *BuildVariant, project *Project, v *Version) (string, []string, error) {
+	distroAliases := []string{}
+	if len(buildVarTask.RunOn) > 0 {
+		distroID := buildVarTask.RunOn[0]
+		if len(buildVarTask.RunOn) > 1 {
+			distroAliases = buildVarTask.RunOn[1:]
+		}
+		return distroID, distroAliases, nil
+	} else if len(buildVariant.RunOn) > 0 {
+		distroID := buildVariant.RunOn[0]
+		if len(buildVariant.RunOn) > 1 {
+			distroAliases = buildVariant.RunOn[1:]
+		}
+		return distroID, distroAliases, nil
+	}
+	grip.Error(message.Fields{
+		"task_id":   id,
+		"message":   "task is not runnable as there is no distro specified",
+		"variant":   buildVariant.Name,
+		"project":   project.Identifier,
+		"version":   v.Revision,
+		"requester": v.Requester,
+	})
+	return "", nil, errors.Errorf("task '%s' is not runnable as there is no distro specified", id)
+}
+
+func shouldRunOnContainer(taskRunOn, buildVariantRunOn []string, containers []Container) task.ExecutionPlatform {
+	containerNameMap := map[string]bool{}
+	for _, container := range containers {
+		containerNameMap[container.Name] = true
+	}
+	runOn := []string{}
```

Comment:
`var runOn []string`
Since `runOn` is assigned to in both the if and else block, Go will complain that `runOn` is initialized to a value but never used because it's unconditionally overwritten.